modules/tinymce/src/core/main/ts/api/Mode.ts

```diff
@@ -86,7 +108,14 @@ export const create = (editor: Editor): EditorMode => {
   registerReadOnlyContentFilters(editor);
   registerReadOnlySelectionBlockers(editor);
 
+  const getReadonlyFromProperty = (property: ReadonlyProperty) => {
+    const mode = availableModes.get()[activeMode.get()].editorReadOnly;
```

Comment:
That's a bit messy. We might need to change the `activeMode` cell to have a bit more information.

I also wonder whether some of this should move into the `mode/Mode.ts` file, the `api/Mode.ts` file is trying to be very simple and avoid the complexities of mode management. That's why `setMode` is over there.

- [ ] I think we can do better, and in particular avoid `Obj.get`

Reply:
I've simplified that and moved to `mode/Mode.ts` instead of placing that in `api/Mode.ts` that should make the api and the function cleaner.